src/triage/component/audition/thresholding.py

```diff
@@ -21,6 +21,66 @@ def _of_metric(df, metric_filter):
     ]
 
 
+class ModelGroupChecker(object):
+    def __init__(
+        self,
+        train_end_times,
+        initial_model_group_ids,
+        models_table,
+        db_engine
+    ):
+        """Before create the distance table, we want to check if the model_group_ids
+        and train_end_times passed in are reasonable to prevent incomparable model groups
+        from being populated to the distance table.
+
+        Args:
+            train_end_times (list) The set of train end times to consider during iteration
+            initial_model_group_ids (list) The initial list of model group ids to narrow down
+            models_table (string) The name of the results schema
+            db_engine (sqlalchemy.engine) A database engine with access to a results schema
+                of a completed modeling run
+        """
+        self.train_end_times = train_end_times
+        self._initial_model_group_ids = initial_model_group_ids
+        self.models_table = models_table
+        self.db_engine = db_engine
+
+    def _check_list_of_string(self, obj):
+        return bool(obj) and all(isinstance(element, str) for element in obj)
+
+    def have_same_train_end_times(self):
+        """Returns the model groups which have the same train end times as the
+        input train end times
+
+        Returns:
+            (set) The model group ids have the same train end times
+        """
+        logging.info("Checking if all model groups have the same train end times")
+        pre_aud = PreAudition(self.db_engine)
+        if self._check_list_of_string(self.train_end_times):
+            end_times_str = self.train_end_times
+        else:
+            end_times_str = [end_time.strftime('%Y-%m-%d') for end_time in self.train_end_times]
+        logging.info(f"Found {len(self._initial_model_group_ids)} total model groups")
+        query = f"""
+           SELECT model_group_id
+           FROM (
+               SELECT model_group_id,
+               array_agg(distinct train_end_time::date::text) as train_end_time_list
+               FROM model_metadata.{self.models_table}
+               WHERE model_group_id in {tuple(self._initial_model_group_ids)}
+               GROUP BY model_group_id
+           ) as t
+           WHERE train_end_time_list = ARRAY{end_times_str}
+           """
+        model_group_ids = pre_aud.get_model_groups(query)
```

Comment:
What are we getting from the PreAudition object here? It looks like we could just run this query through the engine.